### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by walkowif**

### CODE REVIEW

**Review by arkadiuszbeer — APPROVED** [review]
LGTM

**Review by cicdguy — COMMENTED** [review]
@KlaudiaBB - please review.

@walkowif Also update init.sh to remove these files whenever a user wants to use this package template.

**Comment by walkowif** [comment]
@cicdguy Updated `init.sh` accordingly.

**Comment by cicdguy** [comment]
@walkowif Please add `workflows.md` and `images` to `.Rbuildignore`. This will prevent them from going into the package build.

**Comment by walkowif** [comment]
> @walkowif Please add `workflows.md` and `images` to `.Rbuildignore`. This will prevent them from going into the package build.

Done that as well 🙂

**Review by cicdguy — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -68,6 +68,8 @@ All CI/CD jobs are defined in the [.github/workflows](./.github/workflows) direc
 
 All workflows originating from this repository can be repurposed by other R package Github repositories.
```
Comment (KlaudiaBB) [inline README.md:69]:
```suggestion
All workflows originating from this repository can be repurposed by other R package GitHub repositories.
```

File: README.md [hunk 2]
```diff
@@ -68,6 +68,8 @@ All CI/CD jobs are defined in the [.github/workflows](./.github/workflows) direc
...
 
+For detailed description and screenshots of workflows please refer to [this document](./workflows.md).
```
Comment (KlaudiaBB) [inline README.md:71]:
```suggestion
For more information including detailed description and screenshots of workflows, please refer to [Workflows](./workflows.md).
```

File: README.md [hunk 3]
```diff
@@ -68,6 +68,8 @@ All CI/CD jobs are defined in the [.github/workflows](./.github/workflows) direc
...
+
 ### Pre-commit
 
 This repository contains an example [pre-commit] configuration.
```
Comment (KlaudiaBB) [inline README.md:75]: Comment to line 77: [pre-commit] is a tool that uses [Git hooks] to identify and resolve simple issues before submission for code review.

File: workflows.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+This workflow implements Bioconductor-specific R package checks with
+[BiocCheck](https://bioconductor.org/packages/release/bioc/html/BiocCheck.html).
+
+<img src="images/bioccheck.png"  width="40%">
+
+## [`build-check-install.yaml`](./.github/workflows/build-check-install.yaml)
+
+This workflow:
+
+* builds R package,
```
Comment (KlaudiaBB) [inline workflows.md:22]:
```suggestion
1. Build an R package.
```

File: workflows.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+## [`bioccheck.yaml`](./.github/workflows/bioccheck.yaml)
+
+This workflow implements Bioconductor-specific R package checks with
+[BiocCheck](https://bioconductor.org/packages/release/bioc/html/BiocCheck.html).
+
+<img src="images/bioccheck.png"  width="40%">
+
+## [`build-check-install.yaml`](./.github/workflows/build-check-install.yaml)
+
+This workflow:
```
Comment (KlaudiaBB) [inline workflows.md:20]:
```suggestion
This workflow includes the following activities:
```

File: workflows.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+* builds R package,
+* runs `R CMD check`,
```
Comment (KlaudiaBB) [inline workflows.md:23]:
```suggestion
2. Run `R CMD check`.
```

File: workflows.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+* publishes unit test summary,
```
Comment (KlaudiaBB) [inline workflows.md:24]:
```suggestion
3. Publish unit test summary.
```

File: workflows.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+* catches any notes, warnings etc. in the `R CMD check` output,
```
Comment (KlaudiaBB) [inline workflows.md:25]:
```suggestion
4. Catch any notes, warnings etc. in the `R CMD check` output.
```

File: workflows.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+* installs the package.
```
Comment (KlaudiaBB) [inline workflows.md:26]:
```suggestion
5. Install the package.
```

File: README.md (hunk 3 above)
Comment (walkowif) [reply README.md:75]: Updated according to the suggestion.

File: workflows.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+
+## [`gitleaks.yaml`](./.github/workflows/gitleaks.yaml)
+
+This workflow runs [`gitleaks`](https://github.com/zricethezav/gitleaks) on the repo to discover
+any secrets that might have been committed.
+
+<img src="images/gitleaks.png"  width="50%">
+
+Additionally, it runs [`presidio-cli`](https://github.com/insightsengineering/presidio-cli) to find
+any personally identifiable information within the `git` repo.
```
Comment (KlaudiaBB) [inline workflows.md:38]:
```suggestion
any personally identifiable information (PII) within the `git` repo.
```

File: workflows.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+This workflow uses [`roxygen`](https://roxygen2.r-lib.org/) to generate `.Rd` files in
+`man/` directory. It also checks if manuals are up-to-date with roxygen comments in the code.
+
+<img src="images/roxygen.png"  width="80%">
+
+## [`spelling.yaml`](./.github/workflows/spelling.yaml)
+
+Spellchecks are performed by this workflow, and the
+[`spelling`](https://docs.ropensci.org/spelling/) R package is used to detect spelling mistakes.
+In the `inst/WORDLIST` file, you can add words and or acronyms that you want the
```
Comment (KlaudiaBB) [inline workflows.md:102]:
```suggestion
In the `inst/WORDLIST` file, you can add words and/or acronyms that you want the
```

File: workflows.md [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+
+This workflow generates a license report of R package's dependencies for
+continuous compliance.
+
+<img src="images/license-report.png"  width="50%">
+
+## [`links.yaml`](./.github/workflows/links.yaml)
+
+This workflow checks whether URLs embedded in code and documentation are valid. This workflow uses
+[`lychee`](https://github.com/lycheeverse/lychee) to detect broken links. Occasionally this check
```
Comment (KlaudiaBB) [inline workflows.md:61]:
```suggestion
[`lychee`](https://github.com/lycheeverse/lychee) to detect broken links. Occasionally, this check
```

File: workflows.md [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+
+This workflow lints the codebase using [`super-linter`](https://github.com/github/super-linter).
+
+<img src="images/superlinter.png"  width="80%">
+
+## [`pkgdown.yaml`](./.github/workflows/pkgdown.yaml)
+
+Documentation for the R package is generated via this workflow. This workflow uses the
+[`pkgdown`](https://pkgdown.r-lib.org/) framework to generate documentation in HTML,
+and the HTML pages are deployed to the `gh-pages` branch.
```
Comment (KlaudiaBB) [inline workflows.md:77]:
```suggestion
and the HTML pages are then deployed to the `gh-pages` branch.
```

**Pull request merged**